tornjak-frontend/src/components/spiffe-helper.js

```diff
@@ -114,64 +111,108 @@ class SpiffeHelper extends Component {
-          }
+    for (let i = 0; i < agents.length; i++) {
+      let agent = agents[i];
+      let agentId = this.getAgentSpiffeid(agent);
+      agentEntriesDict[agentId] = [];
+
+      let agentSelectors = new Set(agent.selectors.map(formatSelectors));
+      let isAssocWithAgent = e => {
+        let entrySelectors = new Set(e.selectors.map(formatSelectors))
+        if (isSuperset(agentSelectors, entrySelectors)) {
+          agentEntriesDict[agentId].push(e);
+        }
+      };
+      lambdas.push(isAssocWithAgent);
+    }
+
+    for (let i = 0; i < nodeEntries.length; i++) {
+      for (let j = 0; j < lambdas.length; j++) {
+        lambdas[j](nodeEntries[i])
       }
+    }
 
-      return agentEntriesDict
+    return agentEntriesDict
   }
 
   // getCanonicalAgentSpiffeid takes in a agent entry, and returns the first 
   // agent ID that is found associated with it.
   getCanonicalAgentSpiffeid(entry, agents) {
-      let entrySelectors = new Set(entry.selectors.map(formatSelectors));
-      let isAssocWithEntry = a => {
-          let agentSelectors = new Set(a.selectors.map(formatSelectors));
-          return isSuperset(agentSelectors, entrySelectors);
+    let entrySelectors = new Set(entry.selectors.map(formatSelectors));
+    let isAssocWithEntry = a => {
+      let agentSelectors = new Set(a.selectors.map(formatSelectors));
+      return isSuperset(agentSelectors, entrySelectors);
+    }
+
+    let filteredAgents = agents.filter(isAssocWithEntry);
+    if (filteredAgents.length > 0) {
+      return this.getAgentSpiffeid(filteredAgents[0]);
+    }
+    return "";
+  }
+
+  // numberEntries takes in spiffe id of an agent and list of entries
+  // returns number of entries in an agent
+  numberEntries(spiffeid, globalEntries, globalAgents) {
+    if (typeof globalEntries !== 'undefined') {
+      var entriesList = this.getChildEntries(spiffeid, globalAgents, globalEntries);
+      if (typeof entriesList === 'undefined') {
+        return NaN
+      } else {
+        return entriesList.length
       }
+    } else {
+      return NaN
+    }
+  }
 
-      let filteredAgents = agents.filter(isAssocWithEntry);
-      if (filteredAgents.length > 0) {
-          return this.getAgentSpiffeid(filteredAgents[0]);
+  // getChildEntries takes in spiffeid of agent, list of agents, and a list of entries
+  // returns list of child entries
+  getChildEntries(spiffeid, globalAgents, globalEntries) {
+    if (typeof globalEntries === 'undefined') {
+      return NaN;
+    }
+    var curAgent = globalAgents.filter(agent => {
+      return this.getAgentSpiffeid(agent) === spiffeid
+    })
+    var entriesList = globalEntries.filter(entry => {
+      return spiffeid === (this.getEntryParentid(entry))
+    })
+    entriesList = entriesList.concat(this.getAgentEntries(curAgent, globalEntries)); //include node entries
+    // Add entries associated with this agent
+    let agentEntriesDict = this.getAgentsEntries(globalAgents, globalEntries)
```

Comment:
just for one agent you probably want `getAgentEntries` instead of getting it for every agent.

Reply:
done!